### PR TITLE
Allow soft keywords as identifiers

### PR DESCRIPTION
**Pull request by susliko**
Resolves https://github.com/tree-sitter/tree-sitter-scala/issues/227

Problem
-------
Soft keywords ('open', 'end', 'infix', etc.) are not allowed to be used as identifiers, e.g. in
```scala
open()
```
Solution
-------
Introduce `$._soft_identifier` with a lower precedence than modifiers

### CODE REVIEW

**Comment by susliko** [comment]
This one is ready for review

cc @eed3si9n

File: grammar.js [hunk 1]
```diff
@@ -63,6 +62,10 @@ module.exports = grammar({
     $.literal,
   ],
 
+  precedences: $ => [
```
Comment (eed3si9n) [inline grammar.js:65]: Interesting. I wasn't aware of this feature. Could you add a comment linking to a doc (maybe it's just https://github.com/tree-sitter/tree-sitter/pull/939 for now?) for this?

Comment (susliko) [reply grammar.js:65]: Sure!
These relative precedences seem to bring more transparency compared to global numeric precedences

**Review by eed3si9n — APPROVED** [review]
Thanks @susliko!

**Pull request merged**